File: pset2/readability.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>
#include <string.h>
int main(void)
{
    float index = 0;
    float words = 0;
    float sentences = 0;
    float letters = 0;
    string text = get_string("Text:");
    if(strlen(text) > 0)
    {
        words += 1;
    }
    for(int i = 0; i < strlen(text); i++)
    {
        if(text[i] == ' ' || text[i] == '\0')
        {
            words++;
        }
        if((text[i] == '.' || text[i] == '!' || text[i] == '?') && text[i+1] != '.' )
        {
            sentences++;
        }
        if((text[i] >= 65 && text[i] <= 90) || (text[i] >= 97 && text[i] <= 122))
        {
          letters++;
        }

    }
    ///index = 0.0588 * L - 0.296 * S - 15.8
    if(words < 100)
    {
        words = 100/ words;
    }
    if(words > 100)
    {
        words = words / 100;
    }
    index = 0.0588 * (letters * words) - 0.296 * (sentences * words) - 15.8;

    index = round(index);
    int grade = (int)index;
    ///printf("Words: %f\n", words);
    //printf("letters: %f\n", letters);
   // printf("sentences: %f\n", sentences);
    if(grade < 1)
    {
        printf("Before Grade 1\n");
        return 1;
    }
    if(grade >= 16)
    {
        printf("Grade 16+ \n");
        return 1;
    }
    else
    {
        printf("Grade %i\n", grade);

    }
}

File: pset6/readability/readability.py
class text(str):
    def __init__(self):
        self.read = str()
        self.words = float()
        self.letters = float()
        self.sentences = float()
alphabet = list('abcdefghijklmnopqrstuvwxyz')
upper = list('ABCDEFGHIJKLMNOPQRSTUVWXYZ')
punct = list('.!?')

def main():
    tsource = text()
    tsource.read = input("Text:")
    if(len(tsource.read) > 0):
        tsource.words += 1
    for i in range(len(tsource.read)):
        if(tsource.read[i] == ' '):
            tsource.words += 1
        if((tsource.read[i] in alphabet) == True) or ((tsource.read[i] in upper) == True):
            tsource.letters += 1
        if((tsource.read[i] in punct) == True and ((tsource.read[i-1] in punct) == False)):
            tsource.sentences += 1
    """these lines use the in operators and their boolean operators to evaluate every chracter of the user input against some preset global lists"""
    if(tsource.words > 100):
        tsource.words /= 100

    if(tsource.words < 100):
        tsource.words  = 100 / tsource.words

    grade = 0.0588 * (tsource.letters * tsource.words) - 0.296 * (tsource.sentences * tsource.words) - 15.8
    grade = round(grade)

    if(grade >= 16):
        print("Grade 16+\n")
        return 0
    if(grade <= 1):
        print("Before Grade 1\n")
        return 0
    else:
        print("Grade " + str(grade),"\n")
        return 0

main()